tempesta_fw/http_sess.c

```diff
@@ -659,10 +659,12 @@ tfw_http_sess_check_redir_mark(TfwHttpReq *req, RedirMarkVal *mv)
 	unsigned long tmt = HZ * (unsigned long)req->vhost->cookie->tmt_sec;
 
 	if (tfw_http_redir_mark_get(req, &mark_val)) {
-		if (tfw_http_redir_mark_verify(req, &mark_val, mv)
-		    || ++mv->att_no > max_misses
-		    || (tmt && mv->ts + tmt < jiffies))
-		{
+		int r = tfw_http_redir_mark_verify(req, &mark_val, mv);
+
+		if (tmt && mv->ts + tmt < jiffies)
```

Comment:
Possible access out of variable borders. `tfw_http_redir_mark_verify()` can fail because of its size. Rmark can be smaller than `mv->ts`.